### ISSUE TITLE
Use IO-Subgraph + Blockwise throughout codebase

### ISSUE THREAD

**Issue by rjzamora** [dask/dask#6791]
Blocked by #6715 

After #6715 goes in, most (maybe all) IO operations in the `dataframe` and `array` modules can be converted into `Blockwise` high-level graph (HLG) layers.  This change should improve  HLG optimizations (like blockwise fusion).

This issue is intended as a "living" checklist of action items related to `Blockwise` IO...

- [ ] [read_json](https://github.com/dask/dask/blob/f937c06a2ee1b92515714365ea945ca7b4d0e945/dask/dataframe/io/json.py#L94)
- [ ] [array creation functions](https://github.com/dask/dask/blob/master/dask/array/creation.py) (**TODO**: Expand into a specific list)
- [ ] ...

**Comment by rjzamora** [comment]
**Update**: The description above has been modified to include a more-complete list of missing IO routines that may need to be modified to use a Blockwise-IO layer.

cc @mrocklin - Who may have advice on which functions to label as high-priority items.

Before working through this list, it *may* also make sense to introduce a new `BlockwiseIO` layer that is *guaranteed* to include an IO subgraph (while the vanilla `Blockwise` layer will be guaranteed to **not** include any IO).  The fusion of a `BlockwiseIO` layer and `Blockwise` layer will always result in a single `BlockwiseIO layer`... The motivation for this would be to extract the IO complexity introduced by #6715 before initializing the vanilla `Blockwise` layer throughout the code base for IO.

**Comment by jrbourbeau** [comment]
cc-ing @ian-r-rose who might also have some time to work on this

**Comment by jakirkham** [comment]
Is it worth creating a doc page explaining how people go about implementing an HLG operation themselves? Maybe that helps people feel more comfortable filling out gaps in the codebase. Not to mention writing their own in their code

**Comment by jrbourbeau** [comment]
> Is it worth creating a doc page explaining how people go about implementing an HLG operation themselves? 

Cross referencing https://github.com/dask/dask/issues/7709 and https://github.com/dask/dask/issues/7755 for updating documentation around HLGs

**Comment by jakirkham** [comment]
Not sure how feasible this is, but should we consider `rechunk`ing here too?

**Comment by rjzamora** [comment]
The DataFrame component of this issue was superseded by dask-expr/query-planning. My assumption is that Dask Array will also move towards an expression-based API.
